pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -1245,7 +1246,8 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                                 cause = new TopicNotFoundException("Topic Not Found.");
                             }
 
-                            if (!(cause instanceof ServiceUnitNotReadyException)) {
+                            if (!(cause instanceof ServiceUnitNotReadyException)
+                                && (!Exceptions.isExceptionPresentInChain(cause, ManagedLedgerException.class))) {
```

Comment:
The utility method could also take care of checking for `ServiceUnitNotReadyException`

Reply:
good idea. let me update the patch